tests/test_issue2954.py

```diff
@@ -0,0 +1,40 @@
+from django.db import models
+from django.test import TestCase
+from django.core.exceptions import ValidationError
+from rest_framework import serializers
+from rest_framework import viewsets
+from rest_framework.test import APIRequestFactory
+from rest_framework import status
+
+
+factory = APIRequestFactory()
+
+
+class TestInvalidModel(TestCase):
+
+    def test_invalid_model(self):
+        """
+        """
+        class InvalidModel(models.Model):
+            afield = models.CharField(blank=True, max_length=10)
+
+            def save(self, *args, **kwargs):
+                # never save this object.
+                raise ValidationError('a validation error')
```

Comment:
If you're doing that then you'll need to override the serializer.save and catch the case there.